.bazelci/presubmit.yml

```diff
@@ -293,8 +293,6 @@ tasks:
       - "-//tests/core/go_proto_library_importmap:importmap_test"
       - "-//tests/core/go_test:data_test"
       - "-//tests/core/go_test:pwd_test"
-      - "-//tests/core/nogo/coverage:coverage_test"
-      - "-//tests/core/nogo/coverage:gen_code_test"
```

Comment:
I will let @fmeum weigh in here, but I think a benefit of having a list like this is documenting what is supported and not supported. If we move things out of this list by leveraging `target_compatible_with`, then there is no longer a single place acting as the maintainers' TODO list anymore.

Reply:
I agree with @sluongng here. We could go with `target_compatible_with` instead, but since we already have this central list, we should stick to it and maintain it as the single source of truth.

Reply:
Thanks for the suggestion, that makes sense.

fwiw, I was referring to the coverage tests, and I favored this approach as it doesn't limit to CI, but also ensures the incompatible tests won't be triggered on Windows platform if running locally.  https://github.com/bazelbuild/rules_go/blob/d5bc6adb81d3aba03ff625e0c7e57504caf49222/tests/core/coverage/BUILD.bazel#L16-L18

But since the yaml list is the preferred approach, I would love to switch to that.